### PR TITLE
feat(misc): Animation speed/behaviors for button, switches and knobs

### LINKED ISSUE #5602: Increase button/switch/knob animation speed
### Aircraft Version

Development

### Description

The animation of every single button and switch in the cockpit are far too slow to give a realistic feedback of movement. Often the button sound is finished playing while the button is still moving. Especially pushbuttons are very noticable slow. My estimated guess is to speed them up by 3x times and then see how that looks.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

St54Kevin#3583

### PR DESCRIPTION
**Pull request by bouveng**
Fixes #5602

## Summary of Changes
This PR increases the **speed** (between 2-3x) of **all** buttons, switches, knobs and korry buttons.

## Summary for Reviewers
This PR does the following
- Introduces a **utility template** for animation **interactions**, that holds a subset of the different animation properties. This allows for a **single place of configuration**.
- **Splits** the WX radar component into **2 separate templates**, for separation of concerns and allowing of individual animation properties for switches and knob.
- Removes **one state** for the ATC ALT RPTG knob. 2 instead of 3.
- Adds standardized comments where useful, following panel/item naming in cockpit.
- Many changes, but the changes is of the **same sort**, see below.
- PR has **0 impact** on any functionality.

Most changes looks like this, change of template, old template being fed into the new one, and a choice of animation type.
```diff
- <UseTemplate Name="FBW_AIRLINER_Knob_ND_Template">
+ <!-- FCU ND ROSE L -->
+ <UseTemplate Name="FBW_Anim_Interactions">
+     <ANIM_TYPE>KNOB</ANIM_TYPE>
+     <ANIM_TEMPLATE>FBW_AIRLINER_Knob_ND_Template</ANIM_TEMPLATE>
```
## Additional context
It is a lot of buttons. :dizzy_face:

Discord username (if different from GitHub): @bouveng

## Testing instructions
1. Spawn
2. Verify that all buttons "work"
3. Verfy that button, switches, korrys and knobs, are speedy and snappy.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by tracernz** [comment]
The EFIS controls are refactored and rewritten in RNP in the cfms branch. That will clash with this a little. 😆

**Comment by tracernz** [comment]
I haven’t analysed your PR yet. For example the EFIS controls are in a new file with templates for exclusive options etc. https://github.com/flybywiresim/a32nx/blob/autopilot-custom-fpm/src/behavior/src/A32NX_Interior_EFIS.xml

**Comment by bouveng** [comment]
> EFIS panel controls are in a new file with templates for exclusive options etc.

Had a quick and dirty look, and I do not see any problems with using the wrapper in these cases/templates. The utility template can be used as high or low in the tree as one want.

They can also work in parallel.

File: flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml [hunk 1]
```diff
@@ -444,129 +444,167 @@
             <SEQ1_POWERED>1</SEQ1_POWERED>
             <SEQ2_POWERED>1</SEQ2_POWERED>
         </DefaultTemplateParameters>
-
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml:447]: **question:** Why remove the empty lines ?

**Review by Benjozork — CHANGES_REQUESTED** [review]
In general I feel like this should be done a bit differently.

Could we not do something similar to this:

```xml
<Component ID="xyz">
    <UseTemplate Name="FBW_Anim_interactions">
        <ANIM_TYPE>BUTTON</ANIM_TYPE>
    </UseTemplate>

    <UseTemplate Name="FBW_xyz">
        <!-- parameters ... -->
    </UseTemplate>
</Component>
```

Where `FBW_Anim_interactions` would create a `Parameters[Type="Default"]` in the component type ?

File: flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml (hunk 1 above)
Comment (bouveng) [reply flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml:447]: I think I reasoned that the `<!-- comment -->` would be enough spacing. I'm all for spacing and readability, but there is both patterns in the prioject.

Comment (bouveng) [reply flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml:447]: @Benjozork Added back spacing.

**Comment by bouveng** [comment]
Branch is rebased on top of FMGC https://github.com/flybywiresim/a32nx/pull/4953.
Needs a little more work to ensure templating still works and or a few changes in the new FMGC stuff.

**Comment by bouveng** [comment]
@Benjozork PR is now compatible with FMGC #4953. Just small template change.

**Comment by bouveng** [comment]
> EFIS panel controls are in a new file with templates for exclusive options etc. https://github.com/flybywiresim/a32nx/blob/autopilot-custom-fpm/src/behavior/src/A32NX_Interior_EFIS.xml

Support for FMGC #4953 added.
Last commit contains small changes.
Did not need to touch any of the above,

**Comment by bouveng** [comment]
> In general I feel like this should be done a bit differently.
> Could we not do something similar to this:
> 
> ```
> <Component ID="xyz">
>     <UseTemplate Name="FBW_Anim_interactions">
>         <ANIM_TYPE>BUTTON</ANIM_TYPE>
>     </UseTemplate>
> 
>     <UseTemplate Name="FBW_xyz">
>         
>     </UseTemplate>
> </Component>
> ```
> 
> Where `FBW_Anim_interactions` would create a `Parameters[Type="Default"]` in the component type ?

@Benjozork As an answer to that, not everything is a component, and many components use different buttons and switches combinations. So the above wont work in those cases.

**Comment by wpine215** [comment]
Is this PR ready to review?

**Review by Benjozork — APPROVED** [review]

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#5438
Object of testing: #6159 
Tier of Testing: 1
Date: 17/01/22

Testing Process: 
Loaded in C&D and tested every button to see speed and some functionality
Negatives: a few buttons are still slow or have other issues
APU DISCH-slow old animation
Console flood FO-steps are not properly spaced, Captain side is perfect so just look at making them both the same
AP Disconnect on sidestick-slow old animation
Cockpit door button-weirdly comes back up slowly but goes in fast
ATC button on TCAS-seems to have slow old animation but can't tell if its supposed to be like that
Testing Results: Not Passed

Conclusions: Defintely a lot better at this speed. Just found a few issues but otherwise great pr!
Media: 
https://streamable.com/1s6s68

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #6159 
Tier of Testing: 1
Date: 25/01/2022

**Testing Process:**
- Tested all buttons in the cockpit

**Negatives:** 
- Not related but why does the mouse curser disappear while holding down a button click? Never noticed it before but it creates some sort of flickering effect especially when typing fast on the MCDU

**Testing Results:** Passed (brightness knobs are fine for me)

**Comment by bouveng** [comment]
@FlightSimmer7700 

- FIXED: `APU DISCH AGENT` :heavy_check_mark:
- FIXED: `YOKE L AP DISC` :heavy_check_mark:
- FIXED: `YOKE R AP DISC` :heavy_check_mark:
- FIXED: `COCKPIT DOOR INDICATOR` :heavy_check_mark: (not a button anymore, indicator only)
- FIXED: `XPNDR MODE KNOB` :heavy_check_mark:
- NOT FIXED: `F/O CONSOLE/FLOOR LT` :x: (issue with model animation, not animation speed)

**Comment by mico975** [comment]
Quality Assurance Trainee Report

Discord          : mico#3145
Object of testing: #6159 
Tier of Testing  : 1
Date             : 28/01/2022

Testing Process: 
Went thrugh all buttons in the cockpit, all work and are snappy.
All additional fixes are good.

Negatives:       
Release speed of FCU buttons appears a bit slow to me, but not a deal breaker, just personal opinion. 

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#5438
Object of testing: #6159
Tier of Testing: 1
Date: 28/1/22

Testing Process: 
Loaded in C&D and retested most buttons and especially issues found last time
Negatives: N/A
Testing Results: Passed

Conclusions: LGTM

**Pull request merged**